src/vl-map-search.js

```diff
@@ -123,50 +129,112 @@ export class VlMapSearch extends vlElement(HTMLElement) {
       this.__searchEventListenerRegistered = true;
       this._selectElement.addEventListener('search', (event) => {
         if (event && event.detail && event.detail.value) {
-          fetch(this.searchUrl + event.detail.value).then((response) => {
-            return response.json();
-          }).then((data) => {
-            if (data && data.SuggestionResult) {
-              const resultaten = data.SuggestionResult.map((resultaat) => {
-                return {
-                  value: resultaat,
-                  label: resultaat,
-                };
-              });
-              this._selectElement.choices = resultaten;
-            }
-          });
+          const lambertCoordinaat = LambertCoordinaat.of(event.detail.value);
+
+          if (LambertCoordinaat.isValid(lambertCoordinaat)) {
+            this._searchChoicesByLambertCoordinaat(lambertCoordinaat);
+          } else {
+            this._searchChoicesByValue(event.detail.value);
+          }
         }
       });
     }
   }
 
+  _searchChoicesByValue(searchValue) {
+    fetch(new URL(this.searchUrl + searchValue).toString()).then((response) => {
```

Comment:
Searchvalue zou nog met encodeURIComponent encoded moeten worden. Niet met encodeURI, want als iemand op iets met & gaat zoeken, gaat dat niet encoded worden en gaat de server dus denken dat er daarachter een nieuwe query param begint.

`new URL(this.searchUrl + searchValue)` gaat hier trouwens nog altijd gewoon string concatenatie doen (+ evt. valideren als de URL constructor dat doet) ipv. de URL op een slimme manier aan elkaar plakken.